### ISSUE TITLE
Object view in admin returns "object of type 'FakeQuerySet' has no len()"

### ISSUE THREAD

**Issue by kftb** [radiac/django-tagulous#18]
Hi, 

I have a very simple model and tried to implement tagulous. It all works fine except that I cannot access an object with tags within the Admin. It refers to both django-admin code as well as tagulous, that is why I am posting it here. 

```

Environment:


Request Method: GET
Request URL: http://127.0.0.1:8000/admin/organizer/document/1/change/

Django Version: 1.10.1
Python Version: 3.4.3
Installed Applications:
['django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'tagulous',
 'organizer']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware']


Template error:
In template /home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/contrib/admin/templates/admin/change_form.html, error at line 33
   object of type 'FakeQuerySet' has no len()   23 : {% endblock %}
   24 : {% endif %}
   25 : 
   26 : {% block content %}<div id="content-main">
   27 : {% block object-tools %}
   28 : {% if change %}{% if not is_popup %}
   29 :   <ul class="object-tools">
   30 :     {% block object-tools-items %}
   31 :     <li>
   32 :         {% url opts|admin_urlname:'history' original.pk|admin_urlquote as history_url %}
   33 :         <a href=" {% add_preserved_ filters history_url %}" class="historylink">{% trans "History" %}</a>
   34 :     </li>
   35 :     {% if has_absolute_url %}<li><a href="{{ absolute_url }}" class="viewsitelink">{% trans "View on site" %}</a></li>{% endif %}
   36 :     {% endblock %}
   37 :   </ul>
   38 : {% endif %}{% endif %}
   39 : {% endblock %}
   40 : <form {% if has_file_field %}enctype="multipart/form-data" {% endif %}action="{{ form_url }}" method="post" id="{{ opts.model_name }}_form" novalidate>{% csrf_token %}{% block form_top %}{% endblock %}
   41 : <div>
   42 : {% if is_popup %}<input type="hidden" name="{{ is_popup_var }}" value="1" />{% endif %}
   43 : {% if to_field %}<input type="hidden" name="{{ to_field_var }}" value="{{ to_field }}" />{% endif %}


Traceback:

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/core/handlers/exception.py" in inner
  39.             response = get_response(request)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/core/handlers/base.py" in _get_response
  217.                 response = self.process_exception_by_middleware(e, request)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/core/handlers/base.py" in _get_response
  215.                 response = response.render()

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/response.py" in render
  109.             self.content = self.rendered_content

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/response.py" in rendered_content
  86.         content = template.render(context, self._request)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/backends/django.py" in render
  66.             return self.template.render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render
  208.                     return self._render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in _render
  199.         return self.nodelist.render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render
  994.                 bit = node.render_annotated(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render_annotated
  961.             return self.render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/loader_tags.py" in render
  174.         return compiled_parent._render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in _render
  199.         return self.nodelist.render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render
  994.                 bit = node.render_annotated(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render_annotated
  961.             return self.render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/loader_tags.py" in render
  174.         return compiled_parent._render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in _render
  199.         return self.nodelist.render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render
  994.                 bit = node.render_annotated(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render_annotated
  961.             return self.render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/loader_tags.py" in render
  70.                 result = block.nodelist.render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render
  994.                 bit = node.render_annotated(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render_annotated
  961.             return self.render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/loader_tags.py" in render
  70.                 result = block.nodelist.render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render
  994.                 bit = node.render_annotated(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render_annotated
  961.             return self.render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/defaulttags.py" in render
  209.                     nodelist.append(node.render_annotated(context))

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render_annotated
  961.             return self.render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/loader_tags.py" in render
  210.                 return template.render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render
  210.                 return self._render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in _render
  199.         return self.nodelist.render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render
  994.                 bit = node.render_annotated(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render_annotated
  961.             return self.render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/defaulttags.py" in render
  209.                     nodelist.append(node.render_annotated(context))

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render_annotated
  961.             return self.render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/defaulttags.py" in render
  209.                     nodelist.append(node.render_annotated(context))

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render_annotated
  961.             return self.render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/defaulttags.py" in render
  315.                 return nodelist.render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render
  994.                 bit = node.render_annotated(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render_annotated
  961.             return self.render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/defaulttags.py" in render
  315.                 return nodelist.render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render
  994.                 bit = node.render_annotated(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render_annotated
  961.             return self.render(context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render
  1050.         return render_value_in_context(output, context)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/template/base.py" in render_value_in_context
  1028.     value = force_text(value)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/utils/encoding.py" in force_text
  76.                     s = six.text_type(s)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/utils/html.py" in <lambda>
  391.         klass.__str__ = lambda self: mark_safe(klass_str(self))

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/forms/boundfield.py" in __str__
  43.         return self.as_widget()

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/forms/boundfield.py" in as_widget
  101.         return force_text(widget.render(name, self.value(), attrs=attrs))

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/django/forms/boundfield.py" in value
  137.         return self.field.prepare_value(data)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/tagulous/forms.py" in prepare_value
  290.         return super(TagField, self).prepare_value(value)

File "/home/kev/.virtualenvs/papermaster/lib/python3.4/site-packages/tagulous/forms.py" in prepare_value
  163.             if len(value) != 1:

Exception Type: TypeError at /admin/organizer/document/1/change/
Exception Value: object of type 'FakeQuerySet' has no len()

```

**Comment by kftb** [comment]
Just saw that it is not yet adapted for Django 1.10. My bad, but it might be a pointer once it will be migrated.

**Comment by troyshu** [comment]
I experienced the same error, also using Django 1.10. If it helps you or anyone else, I installed the `develop` branch of this fork https://github.com/AASHE/django-tagulous, which has been updated to support Django 1.10. It works fine for me so far, hopefully until the official release of a tagulous version that supports Django 1.10.

First `pip uninstall django-tagulous`.
Then, in your `requirements.txt` file, add `-e git://github.com/AASHE/django-tagulous@develop#egg=django-tagulous`

**Comment by radiac** [comment]
That's brilliant, thanks for your work! I've merged this to the branch `feature/django110` to sort out python 3 support.

**Comment by radiac** [comment]
This is now part of the `develop` branch and will be released as 0.12 soon.

**Comment by troyshu** [comment]
Awesome! And thanks for the amazing library :)
